File: src/main/java/com/university/repositories/PrepaymentReportRepository.java
package com.university.repositories;

import com.university.entities.PrepaymentReport;
import org.springframework.data.jpa.repository.JpaRepository;

import java.time.LocalDate;
import java.util.List;

public interface PrepaymentReportRepository extends JpaRepository<PrepaymentReport, Integer> {

    PrepaymentReport findPrepaymentReportByTravelAllowance_Id(Integer travelAllowanceId);

    List<PrepaymentReport> findPrepaymentReportsByPreparationDateBetween(LocalDate start, LocalDate end);
}

File: src/main/kotlin/com/university/repositories/PrepaymentReportRepository.kt
package com.university.repositories

import com.university.entities.PrepaymentReport
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import java.time.LocalDate

@Repository
interface PrepaymentReportRepository : JpaRepository<PrepaymentReport, Int> {
    fun findPrepaymentReportsByPreparationDateBetween(start: LocalDate, end: LocalDate): List<PrepaymentReport>
}
